From 8451ef5e2937e1ef4cc9f67145c982773310b3de Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Sun, 11 Oct 2015 20:37:29 +0800
Subject: [PATCH] Fix a ctypes-related crash on Arch

"Python integers and Python longs are passed as the platforms default C
int type, their value is masked to fit into the C type." So when they
are big enough a segfault will occur.

Many thanks to lilydjwg.

Change-Id: I2472ed375bf183c39267ea4f2aca5ff09dcea8cd
---
 src/utils/font_utils.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/utils/font_utils.py b/src/utils/font_utils.py
index c13eb21..36d85fa 100644
--- a/src/utils/font_utils.py
+++ b/src/utils/font_utils.py
@@ -30,7 +30,9 @@
 
 # initialize
 libfc.FcInit()
+libfc.FcPatternCreate.restype = c_void_p
 pattern = libfc.FcPatternCreate()
+libfc.FcObjectSetBuild.restype = c_void_p
 objectSet = libfc.FcObjectSetBuild("family", "familylang",
                                    "lang", "spacing", None)
 
@@ -86,6 +88,7 @@ def fontsByLocale(locale):
 
     # get all fonts
     libfc.FcFontList.restype = POINTER(FcFontSet)
+    libfc.FcFontList.argtypes = [c_void_p] * 3
     libfc.FcLangSetGetLangs.restype = POINTER(FcStrSet)
     libfc.FcStrListNext.restype = c_char_p
     libfc.FcPatternFormat.restype = c_char_p
